### PR TITLE
add initial README

### LINKED ISSUE #15: Add a README
Can probably fill the README with the contents from https://docs.google.com/document/d/1UfM3jsYKRYcaeQ-Z2iCOgv-zqis1qGbP20e1xO_K6QM/edit?ts=59765506#

### PR DESCRIPTION
**Pull request by cpanato**
You guys can push changes directly to this branch or let me know. I added this more to help :) please let me know if is not good or useless.

fixes: https://github.com/coreos/ktestutil/issues/15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# KTestutil
```
Comment (xiang90) [inline README.md]: ktestutil (Kubernetes test utilities)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+We make some assumptions on the environment to make it easier for the testing utilities to interact with Kubernetes clusters and the machines’ OS.
+
+### Environment
+  * The cluster is created by bootkube
+  * The cluster is running on CoreOS
+
+### Feature Requirements
+  * Util
+    * Test pod status [Created, Running, Failed, Deleted, etc.]
+    * Test TPR status
```
Comment (xiang90) [inline README.md]: CRD

**Comment by xiang90** [comment]
lgtm. Probably need a second eye on it (i created the content... so i am reviewing my own stuff now)

File: README.md (hunk 1 above)
Comment (cpanato) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (cpanato) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+These projects needs to be tested (or checked) against the Kubernetes API. For example, bootkube needs to verify the Kubernetes cluster is bootstrapped as expected by checking control panel pod manifests; etcd operators needs to verify the etcd clustered it creates are deployed correctly as Kubernetes pods. Some of the projects need to be tested against a faulty Kubernetes cluster. For example, bootkube wants to ensure the multi-node cluster it creates can tolerate a minor disaster; etcd-operator wants to ensure it can handle pod failures. When test fails, we also want to have a easy way to retrieve all relevant logs. We want to ship the plain logs to a cloud storage, so that we can analysis the failure afterwards.
+
+
+## Assumptions
+
+We make some assumptions on the environment to make it easier for the testing utilities to interact with Kubernetes clusters and the machines’ OS.
+
+### Environment
+  * The cluster is created by bootkube
+  * The cluster is running on CoreOS
```
Comment (yifan-gu) [inline README.md]: nit: s/CoreOS/Container Linux

File: README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    * Pods
+    * Node components (Kubelet, docker, etc.)
+    * Master components (API server, controller manager, scheduler, etcd, etc.)
+  * Failure injections
+    * Shutdown machine
+    * Restart machine
+    * Reboot machine
+    * Network partition between machines
+    * Kill pods
+  * Machine
```
Comment (yifan-gu) [inline README.md]: nit: Machine utilities?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+We make some assumptions on the environment to make it easier for the testing utilities to interact with Kubernetes clusters and the machines’ OS.
+
+### Environment
+  * The cluster is created by bootkube
+  * The cluster is running on CoreOS
+
+### Feature Requirements
+  * Util
+    * Test pod status [CRD]
+    * Test TPR status
```
Comment (yifan-gu) [inline README.md]: CRD?

**Comment by yifan-gu** [comment]
@cpanato @xiang90 Couple nits. LGTM

File: README.md (hunk 5 above)
Comment (cpanato) [reply README.md]: updated

File: README.md (hunk 4 above)
Comment (cpanato) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (cpanato) [reply README.md]: done

**Review by yifan-gu — APPROVED** [review]
LGTM, thank you @cpanato !

**Pull request merged**